src/main/java/org/jenkins_ci/plugins/flexible_publish/FlexibleMatrixAggregator.java

```diff
@@ -96,11 +97,19 @@ public boolean endRun(MatrixRun run) throws InterruptedException,
      */
     @Override
     public boolean endBuild() throws InterruptedException, IOException {
+        boolean wholeResult = true;
         for (ConditionalMatrixAggregator cma: aggregatorList) {
-            if (!cma.endBuild()) {
-                return false;
+            try {
+                if (!cma.endBuild()) {
+                    listener.error(String.format("[flexible-publish] aggregation with %s failed", cma.toString()));
+                    wholeResult = false;
+                }
+            } catch (Exception e) {
+                e.printStackTrace(listener.error(String.format("[flexible-publish] aggregation with %s is aborted due to exception", cma.toString())));
```

Comment:
Do you mean about `perform`?  `endBuild` just returns `false` for failures.

I'll update `perform` to handle `AbortException` like Jenkins does for build steps
* https://github.com/jenkinsci/jenkins/blob/jenkins-1.596.1/core/src/main/java/hudson/model/Run.java#L1765
* Outputs the message to build logs.
* Outputs the message and the stacktrace to system logs with the `FINE` level.